build.gradle

```diff
@@ -0,0 +1,248 @@
+                                organization.set("Amazon Web Services")
+                                organizationUrl.set("https://aws.amazon.com")
+                                roles.add("developer")
+                            }
+                        }
+                        scm {
+                            url.set("https://github.com/awslabs/smithy.git")
+                        }
+                    }
+                }
+            }
+        }
+
+        // Don't sign the artifacts if we didn't get a key and password to use.
+        if (project.hasProperty("signingKey") && project.hasProperty("signingPassword")) {
+            signing {
+                useInMemoryPgpKeys(
+                        (String) project.property("signingKey"),
+                        (String) project.property("signingPassword"))
+                sign(publishing.publications["mavenJava"])
+            }
+        }
+    }
+
+    // ==== CheckStyle ====
+    // https://docs.gradle.org/current/userguide/checkstyle_plugin.html
+    apply plugin: "checkstyle"
+    tasks["checkstyleTest"].enabled = false
+
+    // ==== Code coverage ====
+    // https://docs.gradle.org/current/userguide/jacoco_plugin.html
+    apply plugin: "jacoco"
+    // report is always generated after tests run
+    test {
+        finalizedBy jacocoTestReport
+    }
+    // tests are required to run before generating the report
+    jacocoTestReport {
+        dependsOn test
+    }
+    jacocoTestReport {
+        reports {
+            xml.enabled false
+            csv.enabled false
+            html.destination file("$buildDir/reports/jacoco")
+        }
+    }
+
+    // ==== Spotbugs ====
+    // https://plugins.gradle.org/plugin/com.github.spotbugs
+    apply plugin: "com.github.spotbugs"
+    // We don't need to lint tests.
+    tasks["spotbugsTest"].enabled = false
+    // Configure the bug filter for spotbugs.
+    spotbugs {
+        effort = "max"
+        excludeFilterConfig = project.resources.text.fromFile("${project.rootDir}/config/spotbugs/filter.xml")
+    }
+}
+
+// The root project doesn't produce a JAR.
+tasks["jar"].enabled = false
+
+// ==== Javadoc ====
+project.afterEvaluate {
+    tasks.javadoc {
+        title = "Smithy API ${version}"
+        setDestinationDir(file("${project.buildDir}/docs/javadoc/latest"))
+        // Build a consolidated javadoc of all subprojects.
+        source(project.subprojects.stream().map({
+            project(it.path).sourceSets.main.allJava
+        }).collect(Collectors.toList()))
+    }
+}
+
+/*
+ * Sonatype Staging Finalization
+ * ====================================================
+ *
+ * When publishing to Maven Central, we need to close the staging
+ * repository and release the artifacts after they have been
+ * validated. This configuration is for the root project because
+ * it operates at the "group" level.
+ */
+if (project.hasProperty("sonatypeUser") && hasProperty("sonatypePassword")) {
```

Comment:
Should be `project.hasProperty("sonatypePassword")`

Reply:
Fixed